crates/burn-dataset/src/dataset/dataframe.rs

```diff
@@ -0,0 +1,390 @@
+
+    use super::*;
+
+    #[derive(Clone, Debug, Deserialize, PartialEq)]
+    struct TestData {
+        a: i32,
+        b: bool,
+        c: f64,
+        d: String,
+        e: Vec<i32>
+    }
+
+    fn create_test_dataframe() -> DataFrame {
+        let s0 = Series::new("a", &[1, 2, 3, ]);
+        let s1 = Series::new("b", &[true, false, true]);
+        let s2 = Series::new("c", &[1.1, 2.2, 3.3]);
+        let s3 = Series::new("d", &["Boo", "Boo2", "Boo3"]);
+        let s4 = Series::new("e", &[
+            vec![1, 2, 3],
+            vec![4, 5, 6],
+            vec![7, 8, 9]
+        ]);
+        DataFrame::new(vec![s1, s2, s0, s3, s4]).unwrap()
+    }
+
+    fn create_test_lazyframe() -> LazyFrame {
+        let s0 = Series::new("a", &[1, 2, 3, ]);
+        let s1 = Series::new("b", &[true, false, true]);
+        let s2 = Series::new("c", &[1.1, 2.2, 3.3]);
+        let s3 = Series::new("d", &["Boo", "Boo2", "Boo3"]);
+        let s4 = Series::new("e", &[
+            vec![1, 2, 3],
+            vec![4, 5, 6],
+            vec![7, 8, 9]
+        ]);
+        DataFrame::new(vec![s1, s2, s0, s3, s4]).unwrap().lazy()
+    }
+
+    #[test]
+    fn test_dataframe_dataset() {
+        let df = create_test_dataframe();
+        let dataset: DataframeDataset<TestData> = DataframeDataset::from_dataframe(df).unwrap();
+
+        assert_eq!(dataset.len(), 3);
+        assert_eq!(dataset.is_empty(), false);
+
+        let item = dataset.get(1).unwrap();
+        assert_eq!(
+            item,
+            TestData {
+                a: 2,
+                b: false,
+                c: 2.2,
+                d: "Boo2".to_string(),
+                e: vec![4, 5, 6]
+            }
+        );
+
+        let item = dataset.get(2).unwrap();
+
+        assert_eq!(
+            item,
+            TestData {
+                a: 3,
+                b: true,
+                c: 3.3,
+                d: "Boo3".to_string(),
+                e: vec![7, 8, 9]
+            }
+        );
+    }
+
+    #[test]
+    fn test_lazyframe_dataset() {
+        let df = create_test_lazyframe();
+        let mut dataset: DataframeDataset<TestData> = DataframeDataset::from_lazyframe(df).unwrap();
+        dataset.collect_and_get_len().unwrap();
+
+        assert_eq!(dataset.len(), 3);
+        assert_eq!(dataset.is_empty(), false);
+
+        let item = dataset.get(1).unwrap();
+        assert_eq!(
+            item,
+            TestData {
+                a: 2,
+                b: false,
+                c: 2.2,
+                d: "Boo2".to_string(),
+                e: vec![4, 5, 6],
+            }
+        );
+
+        let item = dataset.get(2).unwrap();
+
+        assert_eq!(
+            item,
+            TestData {
+                a: 3,
+                b: true,
+                c: 3.3,
+                d: "Boo3".to_string(),
+                e: vec![7, 8, 9],
+            }
+        );
+    }
+
+    // TODO test non existing struct fields
```

Comment:
Missing this TODO. You should test have some data that's not present in your struct fields.